service/internal/changedetection/tccpn.go

```diff
@@ -60,10 +68,39 @@ func (t *TCCPN) ShouldUpdate(ctx context.Context, cr infrastructurev1alpha2.AWSC
 		}
 	}
 
+	var currentRelease releasev1alpha1.Release
+	{
+		currentRelease, err = t.releases.Release(ctx, cc.Status.TenantCluster.ReleaseVersion)
+		if releases.IsNotFound(err) {
+			// fall through
+		} else if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+
+	var targetRelease releasev1alpha1.Release
+	{
+		targetRelease, err = t.releases.Release(ctx, key.ReleaseVersion(&cr))
+		if releases.IsNotFound(err) {
+			// fall through
+		} else if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+
+	componentVersionsEqual := releaseComponentsEqual(currentRelease, targetRelease)
 	masterInstanceEqual := cc.Status.TenantCluster.TCCPN.InstanceType == key.ControlPlaneInstanceType(cr)
 	masterReplicasEqual := cc.Status.TenantCluster.TCCPN.MasterReplicas == rep
 	operatorVersionEqual := cc.Status.TenantCluster.OperatorVersion == key.OperatorVersion(&cr)
 
+	if !componentVersionsEqual {
+		t.logger.LogCtx(ctx,
+			"level", "debug",
+			"message", "detected TCCP stack should update",
+			"reason", "component versions changed",
```

Comment:
Which one changed would be superb.

Reply:
added a func which prints the diff of components